cmd/ipdr/main.go

```diff
@@ -137,23 +141,38 @@ More info: https://github.com/miguelmota/ipdr`,
 		Short: "Start IPFS-backed Docker registry server",
 		Long:  "Start the IPFS-backed Docker registry server that proxies images stored on IPFS to Docker registry format",
 		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := ensureCIDStorePath(cidStorePath); err != nil {
+				return err
+			}
+
 			srv := server.NewServer(&server.Config{
-				Port:        port,
-				Debug:       !silent,
-				IPFSGateway: ipfsGateway,
-				TLSKeyPath:  tlsKeyPath,
-				TLSCertPath: tlsCertPath,
+				Port:         port,
+				Debug:        !silent,
+				IPFSHost:     ipfsHost,
+				IPFSGateway:  ipfsGateway,
+				CIDResolvers: cidResolvers,
+				CIDStorePath: cidStorePath,
+				TLSKeyPath:   tlsKeyPath,
+				TLSCertPath:  tlsCertPath,
 			})
 
 			return srv.Start()
 		},
 	}
 
+	defaultCIDStore, _ := os.UserHomeDir()
+	if defaultCIDStore != "" {
+		defaultCIDStore = filepath.Join(defaultCIDStore, ".ipdr/cids")
+	}
+
 	serverCmd.Flags().BoolVarP(&silent, "silent", "s", false, "Silent flag suppresses logs")
 	serverCmd.Flags().UintVarP(&port, "port", "p", 5000, "The port for the Docker registry to listen on")
 	serverCmd.Flags().StringVarP(&tlsCertPath, "tlsCertPath", "", "", "The path to the .crt file for TLS")
 	serverCmd.Flags().StringVarP(&tlsKeyPath, "tlsKeyPath", "", "", "The path to the .key file for TLS")
+	serverCmd.Flags().StringVarP(&ipfsHost, "ipfs-host", "", "127.0.0.1:5001", "A remote IPFS API host to pull the image from. Eg. 127.0.0.1:5001")
 	serverCmd.Flags().StringVarP(&ipfsGateway, "ipfs-gateway", "g", "127.0.0.1:8080", "The readonly IPFS Gateway URL to pull the image from. Eg. https://ipfs.io")
+	serverCmd.Flags().StringArrayVar(&cidResolvers, "cid-resolver", []string{"file:" + defaultCIDStore, "oci.dhnt.io"}, "Map repo:reference to CID. Accepts dnslink, IPFS path, and local file path.")
```

Comment:
What is `oci.dhnt.io`?

Reply:
What do you think about only having `[]string{"/ipfs"}` as the default here?

Reply:
A domain  dedicated to  resolving dnslink (https://dnslink.io/ ) for publicly sharing docker image name[:tag] -> cid mapping on IPFS.
I own the tld.

Reply:
Ah I see.
Is the dnslink value a hash of the local cids directory? or how do you manage or update the cid resolver domain?
For example, this is how I tested it out:

Pull and tag hello world image:

```bash
docker pull hello-world
docker tag hello-world docker.local:5000/hello-world
docker push --quiet docker.local:5000/hello-world
```

Tree of cids directory contents:

```bash
$ tree ~/.ipdr/
/home/mota/.ipdr/
└── cids
    └── hello-world
        └── latest
```

Add cids directory to IPFS:

```bash
$ ipfs add -r ~/.ipdr/cids/ --quieter
QmVtjwa3kdFJHce2wnFuygaCHdSPXraBd4FRSEZKjqZWQp
```

Set _dnslink TXT record and verify:

```bash
$ dig _dnslink.ipdr.io -t TXT +short
"dnslink=/ipfs/QmVtjwa3kdFJHce2wnFuygaCHdSPXraBd4FRSEZKjqZWQp"
```

Run server with resolver domain:

```bash
go run cmd/ipdr/ipdr.go server --cid-resolver=ipdr.io
```

Run ipdr dig to get cid using repo tag name:

```bash
$  ipdr dig hello-world:latest
bafybeiakvswzlopeu573372p5xry47tkc2hhcg5q5rulmbfrnkecrbnt3y
```

We could use `ipdr.io` domain as a canonical resolver

Reply:
Yes, it is the v1 hash of the top directory. 
`ipfs add -r --cid-version 1 cids`

Reply:
May I also suggest resolve ` local.ipdr.io`  (or similar) to localhost host address `127.0.0.1` as a convenience?
This way we could skip the step of editing the /etc/hosts.

Reply:
@qiangli good idea. It's supported now:

```bash
$ dig local.ipdr.io +short
127.0.0.1
```

Reply:
Moved the repository to an organization: https://github.com/ipdr/ipdr

You should receive an invite to be a collaborator on repo : )

@qiangli

Reply:
thanks!